File: iOS/PanoAudioChat/PanoAudioChatSDK/Source/Sections/AudioRoom/View/Nav/PanoNaviTitleView.m
//
//  PanoNaviTitleView.m
//  PanoAudioChatSDK
//
//  Created by pano on 2021/7/19.
//

#import "PanoNaviTitleView.h"
#import "PanoDefine.h"

@implementation PanoNaviTitleView {
    NSString *title;
    NSString *subTitle;
}

- (instancetype)initWithTitle:(NSString *)title
                    subTitle:(NSString *)subTitle {
    self = [super init];
    if (self) {
        self->title =title;
        self->subTitle = subTitle;
    }
    return self;
}

- (UIView *)naviTitleView {
    UILabel *titleView = [UILabel new];
    titleView.textAlignment = NSTextAlignmentCenter;
    titleView.numberOfLines = 0;
    UIColor *whiteColor = [UIColor whiteColor];
    NSMutableAttributedString *attrStr = [[NSMutableAttributedString alloc] initWithString:title attributes:@{
        NSForegroundColorAttributeName : whiteColor,
        NSFontAttributeName: fontLarger()
    }];
    [attrStr appendAttributedString:[[NSAttributedString alloc] initWithString:@"\n"]];
    NSAttributedString *subStr = [[NSAttributedString alloc] initWithString:subTitle attributes:@{
        NSForegroundColorAttributeName : whiteColor,
        NSFontAttributeName: fontMin()
    }];
    [attrStr appendAttributedString:subStr];
    titleView.attributedText = attrStr;
    return titleView;
}

@end

File: iOS/PanoAudioChat/PanoAudioChat/Classes/Sections/AudioRoom/View/Nav/PanoNaviTitleView.swift
//
//  PanoNaviTitleView.swift
//  PanoAudioChat
//
//  Created by wenhua.zhang on 2020/10/30.
//

import UIKit

class PanoNaviTitleView  {

    var titleView: UILabel!
    var title: String!
    var subTitle: String!
    
    init(title: String, subTitle: String) {
        self.title = title
        self.subTitle = subTitle
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func naviTitleView() -> UIView {
        
        titleView = UILabel()
        titleView.textAlignment = .center
        titleView.numberOfLines = 0
        let whiteColor = UIColor.white
        let attrStr =  NSMutableAttributedString(string: title, attributes: [NSAttributedString.Key.foregroundColor : whiteColor, NSAttributedString.Key.font : FontLarger
                                                                 
        ])
        
        attrStr.append(NSAttributedString(string: "\n"))
        
        let subStr = NSAttributedString(string: subTitle, attributes: [NSAttributedString.Key.foregroundColor : whiteColor, NSAttributedString.Key.font : FontMin
        ])
        
        attrStr.append(subStr)
        
        titleView.attributedText = attrStr
        
        return titleView
    }
}
